experimental/dds/tree2/src/feature-libraries/memoizedIdRangeAllocator.ts

```diff
@@ -11,23 +11,40 @@ import {
 	setInRangeMap,
 	Mutable,
 	brand,
+	generateStableId,
 } from "../util";
 
 /**
- * An unique ID allocator that returns the output ID for the same input ID.
- * "The same" here includes cases where a prior call allocated a range of IDs that partially or fully overlap with the
- * current call.
- * @alpha
+ * A unique ID allocator that returns the output ID for the same input ID.
  *
- * @param revision - The revision associated with the range of IDs to allocate.
- * @param startId - The first ID to allocate.
- * @param count - The number of IDs to allocate. Interpreted as 1 if undefined.
+ * @alpha
  */
-export type MemoizedIdRangeAllocator = (
-	revision: RevisionTag | undefined,
-	startId: ChangesetLocalId,
-	count?: number,
-) => IdRange[];
+export interface MemoizedIdRangeAllocator {
+	/**
+	 * The next ID to allocate.
+	 */
+	nextId: number;
+	/**
+	 * A unique ID allocator that returns the output ID for the same input ID.
+	 *
+	 * "The same" here includes cases where a prior call allocated a range of IDs that partially or fully overlap with the
+	 * current call.
+	 * @param revision - The revision associated with the range of IDs to allocate.
+	 * @param startId - The first ID to allocate.
+	 * @param count - The number of IDs to allocate. Interpreted as 1 if undefined.
+	 */
+	allocate(
+		revision: RevisionTag | undefined,
+		startId: ChangesetLocalId,
+		count?: number,
+	): IdRange[];
+	/**
+	 * Allocates a new range of IDs starting from the nextId.
+	 *
+	 * @param count - The number of IDs to allocate. Interpreted as 1 if undefined.
+	 */
+	mint(count?: number): IdRange[];
```

Comment:
This can just return the first ID since the allocator can always create a single ID range for the set of IDs being minted.